clang/lib/CodeGen/CGStmt.cpp

```diff
@@ -2626,14 +2629,20 @@ void CodeGenFunction::EmitAsmStmt(const AsmStmt &S) {
   SmallVector<TargetInfo::ConstraintInfo, 4> OutputConstraintInfos;
   SmallVector<TargetInfo::ConstraintInfo, 4> InputConstraintInfos;
 
+  const FunctionDecl *FD = dyn_cast_or_null<FunctionDecl>(CurCodeDecl);
```

Comment:
```suggestion
  const auto *FD = dyn_cast_if_present<FunctionDecl>(CurCodeDecl);
```
`dyn_cast_or_null` is deprecated (also, no need to spell out the type here since it’s already evident from the cast).

Reply:
Where do you get dyn_cast_or_null is deprecated? I only remember a rejected RFC to start using dyn_cast_if_present

Reply:
> Where do you get dyn_cast_or_null is deprecated?

Mainly from here (as well as other places in `Casting.h`):
https://github.com/llvm/llvm-project/blob/5cd0ba30f53d11835dbfd05ad4071d397387fb04/llvm/include/llvm/Support/Casting.h#L756-L769

Reply:
I think we should just get rid of dyn_cast_if_present then: https://discourse.llvm.org/t/psa-swapping-out-or-null-with-if-present/65018/12

Reply:
Yeah, I agree it seems weird to keep both around, and just outright removing one of them shouldn’t be too disruptive imo considering that it’s a fairly straight-forward change; I don’t know the history behind this too well, but do we want an RFC to decide on which one to remove, or is there already a clear favourite?

For what it’s worth, we currently have (at least according to grep, however accurate that may be) around 7900 occurrences of calls to functions ending with`cast_or_null` and around 530 ending with `cast_if_present` in the entire codebase.

Reply:
> do we want an RFC to decide on which one to remove

I could look into that if that would help.